clients/src/test/java/org/apache/kafka/common/network/ChannelBuildersTest.java

```diff
@@ -79,25 +80,52 @@ public void testChannelBuilderConfigs() {
 
         // test configs with listener prefix
         Map<String, Object> configs = ChannelBuilders.channelBuilderConfigs(securityConfig, new ListenerName("listener1"));
+
         assertNull(configs.get("listener.name.listener1.gssapi.sasl.kerberos.service.name"));
+        assertFalse(securityConfig.unused().contains("listener.name.listener1.gssapi.sasl.kerberos.service.name"));
+
         assertEquals(configs.get("gssapi.sasl.kerberos.service.name"), "testkafka");
+        assertFalse(securityConfig.unused().contains("gssapi.sasl.kerberos.service.name"));
+
         assertEquals(configs.get("sasl.kerberos.service.name"), "testkafkaglobal");
+        assertFalse(securityConfig.unused().contains("gssapi.sasl.kerberos.service.name"));
+
         assertNull(configs.get("listener.name.listener1.sasl.kerberos.service.name"));
+        assertFalse(securityConfig.unused().contains("listener.name.listener1.sasl.kerberos.service.name"));
 
         assertNull(configs.get("plain.sasl.server.callback.handler.class"));
+        assertFalse(securityConfig.unused().contains("plain.sasl.server.callback.handler.class"));
+
         assertEquals(configs.get("listener.name.listener1.gssapi.config1.key"), "custom.config1");
+        assertFalse(securityConfig.unused().contains("listener.name.listener1.gssapi.config1.key"));
+
         assertEquals(configs.get("custom.config2.key"), "custom.config2");
+        assertFalse(securityConfig.unused().contains("custom.config2.key"));
 
         // test configs without listener prefix
+        securityConfig = new TestSecurityConfig(props);
```

Comment:
Do we need to instantiate again?

Reply:
We need a new ```RecordingMap``` to test different key without listener prefix. Otherwise, the key may be used by previous test.